src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,708 @@
+            comp.VerifyDiagnostics(
+                // (14,15): error CS8915: The delegate type could not be inferred.
+                //         d = p.F2;
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F2").WithLocation(14, 15));
+        }
+
+        [Fact]
+        public void ExtensionMethods_02()
+        {
+            var source =
+@"static class E
+{
+    internal static void F(this System.Type x, int y) { }
+    internal static void F(this string x) { }
+}
+class Program
+{
+    static void Main()
+    {
+        System.Delegate d;
+        d = typeof(Program).F;
+        d = """".F;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ExtensionMethods_03()
+        {
+            var source =
+@"using N;
+namespace N
+{
+    static class E1
+    {
+        internal static void F1(this object x, int y) { }
+        internal static void F2(this object x, int y) { }
+        internal static void F2(this object x) { }
+        internal static void F3(this object x) { }
+    }
+}
+static class E2
+{
+    internal static void F1(this object x) { }
+}
+class Program
+{
+    static void Main()
+    {
+        System.Delegate d;
+        var p = new Program();
+        d = p.F1;
+        d = p.F2;
+        d = p.F3;
+        d = E1.F1;
+        d = E2.F1;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (22,15): error CS8915: The delegate type could not be inferred.
+                //         d = p.F1;
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F1").WithLocation(22, 15),
+                // (23,15): error CS8915: The delegate type could not be inferred.
+                //         d = p.F2;
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "F2").WithLocation(23, 15));
+        }
+
+        [Fact]
+        public void ExtensionMethods_04()
+        {
+            var source =
+@"static class E
+{
+    internal static void F1(this object x, int y) { }
+}
+static class Program
+{
+    static void F2(this object x) { }
+    static void Main()
+    {
+        System.Delegate d;
+        d = E.F1;
+        d = F2;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ExtensionMethods_05()
+        {
+            var source =
+@"static class E
+{
+    internal static void F1<T>(this object x, T y) { }
+    internal static void F2<T, U>(this T t) { }
+}
+class Program
+{
+    static void F<T>(T t) where T : class
+    {
+        System.Delegate d;
+        d = t.F1<int>;
+        d = t.F1<T>;
+        d = t.F2<T, object>;
+        d = t.F2<object, T>;
```

Comment:
Do we have a test similar to this (instance or extension method with type parameters converted to `Delegate`) but where type arguments are not provided on the method group? I assume then the method group will just be empty, is that right?

Reply:
What was the resolution?

---
In reply to: [612862173](https://github.com/dotnet/roslyn/pull/52448#discussion_r612862173) [](ancestors = 612862173)

Reply:
> What was the resolution?

I've updated this test to include method groups without type arguments. It'll be in the next commit.